### PR TITLE
Allow controlling the number of total shards

### PR DESCRIPTION
**Pull request by HopeBaron**
preferably merge after #194

### CODE REVIEW

File: core/src/main/kotlin/ClientResources.kt [hunk 1]
```diff
@@ -11,6 +11,7 @@ class ClientResources(
     val httpClient: HttpClient,
     val defaultStrategy: EntitySupplyStrategy<*>,
     val intents: Intents,
+    val totalShards: Int = shardCount
```
Comment (BartArys) [inline core/src/main/kotlin/ClientResources.kt:14]: We might as well store the complete shard info in the resources, right?

File: core/src/main/kotlin/behavior/GuildBehavior.kt [hunk 2]
```diff
@@ -192,7 +192,7 @@ interface GuildBehavior : KordEntity, Strategizable {
      */
     val gateway: Gateway?
         get() {
-            val shard = id.value.shr(22) % kord.resources.shardCount.coerceAtLeast(1)
+            val shard = id.value.shr(22) % kord.resources.shards.indices.count().coerceAtLeast(1)
```
Comment (hakiobo) [inline core/src/main/kotlin/behavior/GuildBehavior.kt:195]: Shouldn't we be calculating the shard based on the mod by totalShards instead? 
```suggestion
            val shard = id.value.shr(22) % kord.resources.shards.totalShards.coerceAtLeast(1)
```

File: core/src/main/kotlin/ClientResources.kt [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
     val token: String,
-    val shardCount: Int,
+    val shards: Shards,
     val httpClient: HttpClient,
     val defaultStrategy: EntitySupplyStrategy<*>,
     val intents: Intents,
 ) {
     override fun toString(): String {
-        return "ClientResources(shardCount=$shardCount, httpClient=$httpClient, defaultStrategy=$defaultStrategy, intents=$intents)"
+        return "ClientResources(shardCount=$shards, httpClient=$httpClient, defaultStrategy=$defaultStrategy, intents=$intents)"
```
Comment (hakiobo) [inline core/src/main/kotlin/ClientResources.kt:16]:
```suggestion
        return "ClientResources(shards=$shards, httpClient=$httpClient, defaultStrategy=$defaultStrategy, intents=$intents)"
```

File: core/src/main/kotlin/behavior/GuildBehavior.kt (hunk 2 above)
Comment (HopeBaron) [reply core/src/main/kotlin/behavior/GuildBehavior.kt:195]: Good catch

File: core/src/main/kotlin/ClientResources.kt (hunk 3 above)
Comment (HopeBaron) [reply core/src/main/kotlin/ClientResources.kt:16]: There is no need for this, sharding is a data class.

**Review by BartArys — APPROVED** [review]

File: core/src/main/kotlin/ClientResources.kt (hunk 3 above)
Comment (HopeBaron) [reply core/src/main/kotlin/ClientResources.kt:16]: mb, I mis-understood the point of the change

**Pull request merged**